### PR TITLE
docopts: init at 0.6.3-rc2

### PR DESCRIPTION
**Pull request by con-f-use**
###### Motivation for this change
[Docopts](https://github.com/docopt/docopts), mind the trailing "s", is the shell scripting tool for the well known [docopt python](https://github.com/docopt/docopt) library. Written in go.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by con-f-use** [comment]
@SuperSandro2000 @AndersonTorres @zimbatm
Can one of you gents please review/merge this? Sorry to bother you, you're just the first people that I could think of.

**Comment by AndersonTorres** [comment]
Looks cute to me.

File: pkgs/development/tools/misc/docopts/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/docopts/default.nix:3]: Did you try using buildGoModule? It does not require the deps.nix and is preferred.

File: pkgs/development/tools/misc/docopts/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 = "sha256-PmsTkPT/sf70MKYLhHvjCDb2o3VQ1k7d++RUW7rcoAg=";
+  };
+
+  goPackagePath = "github.com/docopt/${pname}";
+
+  goDeps = ./deps.nix;
+
+  postInstall = ''
+    install -D -m 755 ./go/src/$goPackagePath/docopts.sh $out/bin/docopts.sh
+    rm $out/bin/json_t
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/docopts/default.nix:20]: Please only build the binaries we need. I think you can achieve this with:

```
subPackages = [ "docopt" ];
```

File: pkgs/development/tools/misc/docopts/default.nix (hunk 1 above)
Comment (con-f-use) [reply pkgs/development/tools/misc/docopts/default.nix:3]: I did, the source in an old-style go project, that doesn't have a mod file, so we're stuck with GoPackage.

File: pkgs/development/tools/misc/docopts/default.nix (hunk 2 above)
Comment (con-f-use) [reply pkgs/development/tools/misc/docopts/default.nix:20]: Done.

Btw. you suggestion just gave me _'cannot find package "github.com/docopt/docopts/docopts"', I used ` subPackages = [ "./" ];`.

**Comment by SuperSandro2000** [comment]
@ofborg build

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/development/tools/misc/docopts/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  goPackagePath = "github.com/docopt/${pname}";
+
+  goDeps = ./deps.nix;
+
+  subPackages = [ "./" ];
+
+  postInstall = ''
+    install -D -m 755 ./go/src/$goPackagePath/docopts.sh $out/bin/docopts.sh
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/docopts/default.nix:21]: Scripts under bin/ should not end with the file extension.

**Comment by SuperSandro2000** [comment]
Also you forgot to add this to all packages.

**Comment by con-f-use** [comment]
Stupid me, sorry. Good catch!

File: pkgs/development/tools/misc/docopts/default.nix (hunk 3 above)
Comment (con-f-use) [reply pkgs/development/tools/misc/docopts/default.nix:21]: In this case it is meant to be included in other shell scripts, so the `.sh` is sensible, actually.

**Comment by AndersonTorres** [comment]
@con-f-use it is bugged. Read ofborg's log.

**Comment by con-f-use** [comment]
@AndersonTorres, do you mean the following log:
```
nix-env failed:
warning: ignoring the user-specified setting 'restrict-eval', because it is a restricted setting and you are not a trusted user
error: while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/.gc-of-borg-outpaths.nix:44:12, called from undefined position:
while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/top-level/release-lib.nix:121:6, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/lib/attrsets.nix:292:43:
while evaluating 'hydraJob' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/lib/customisation.nix:171:14, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/top-level/release-lib.nix:121:14:
while evaluating the attribute 'drvPath' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/lib/customisation.nix:188:13:
while evaluating the attribute 'drvPath' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/lib/customisation.nix:155:13:
while evaluating the attribute 'buildInputs' of the derivation 'cutelyst-2.13.0' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/stdenv/generic/make-derivation.nix:192:11:
while evaluating the attribute 'UWSGICONFIG_PHP_LDFLAGS' of the derivation 'uwsgi-2.0.19.1' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/stdenv/generic/make-derivation.nix:192:11:
attempt to call something which is not a function but a string, at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/servers/uwsgi/default.nix:97:29
```

That seems like and ofborg problem rather than one with my packages, but admittedly I don't understand what's going on here.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by AndersonTorres** [comment]
All OK?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 106917` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>docopts</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> All OK?

I think so. I am just taking it slowly.

**Comment by con-f-use** [comment]
> I think so. I am just taking it slowly.

And still too fast for me. I would have squashed the commit history after a final "okay". Not that important. Thanks for your hard work again!

**Pull request merged**